fvm/evm/emulator/emulator.go

```diff
@@ -156,12 +157,43 @@ func (bl *BlockView) RunTransaction(
 
 	// update tx context origin
 	proc.evm.TxContext.Origin = msg.From
-	res, err = proc.run(msg, txHash, 0, tx.Type())
+	res, err = proc.run(msg, txHash, txIndex, tx.Type())
 	if err != nil {
 		return nil, err
 	}
 	// all commmit errors (StateDB errors) has to be returned
-	return res, proc.commitAndFinalize()
+	if err := proc.commit(finalize); err != nil {
+		return nil, err
+	}
+
+	// we can always reset after committing,
+	// this clears state for any subsequent transaction runs
+	proc.state.Reset()
+
+	return res, nil
+}
+
+// RunTransaction runs an evm transaction
+func (bl *BlockView) RunTransaction(
+	tx *gethTypes.Transaction,
+) (*types.Result, error) {
+	return bl.runTransaction(tx, true, 0)
+}
+
+func (bl *BlockView) BatchRunTransactions(txs []*gethTypes.Transaction) ([]*types.Result, error) {
+	batchLen := len(txs)
+	res := make([]*types.Result, batchLen)
+
+	for i, tx := range txs {
+		finalize := i+1 == batchLen // only finalize during last transaction
+		r, err := bl.runTransaction(tx, finalize, uint(i))
+		if err != nil {
+			return nil, err
+		}
+		res[i] = r
```

Comment:
After each tx run couple of things has to happen: 
- Commit (if successful or failed)
- Collect Logs (if successful or failed)
- Reset (always)